sitestats/sitestats-impl/src/java/org/sakaiproject/sitestats/impl/StatsAggregateJobImpl.java

```diff
@@ -310,16 +310,20 @@ private String startJob() throws SQLException {
 						if(firstEventIdProcessedInBlock == -1)
 							firstEventIdProcessedInBlock = lastProcessedEventId;
 						processedCounter++;
+						if (event == "pres.end" || event == "pres.begin") {
```

Comment:
this is wrong. you need to compare strings ....

Reply:
event is a string though, am I missing something?

Reply:
In Java, the “==” operator checks if both variables reference the exact same object in memory rather than comparing their character content, so even if two strings look identical, they may not be considered equal if they’re distinct objects; in contrast, PHP’s “==” operator compares the actual values of the strings, which means it returns true if the characters match. This fundamental difference is why a comparison that might work in PHP by simply using “==” fails in Java, where you should instead use the .equals() method to properly compare the content of the strings.

https://stackify.com/equality-in-java-operators-methods-and-what-to-use-when/

Reply:
Right, equals is way safer here. I didn't realize you were commenting on that @ottenhoff . @st-manu, could you use `StringUtils.equalsAny(event, StatsManager.SITEVISIT_EVENTID, StatsManager.SITEVISITEND_EVENTID)` here?